// freyja-ui/esm5/lib/accordion/accordion.component.js
import * as tslib_1 from "tslib";
// Angular Imports
import { Component, Input, Output, EventEmitter } from '@angular/core';
// Internal Imports
import { BaseComponent } from '../shared/base.component';
/**
 * Counter for Next Id
 */
var nextId = 0;
/**
 * Freyja Accordion Component according to specifics of Freyja Design
 */
var AccordionComponent = /** @class */ (function (_super) {
    tslib_1.__extends(AccordionComponent, _super);
    function AccordionComponent() {
        var _this = _super.call(this) || this;
        // This is to make component open state controlled
        _this.isOpen = false;
        /**
         * Provide unique ID or it will use auto generated from counter
         */
        _this.id = "fj-accordion-" + ++nextId;
        /**
         * Will emit if open state get's changed on click
         */
        _this.change = new EventEmitter(false);
        return _this;
    }
    Object.defineProperty(AccordionComponent.prototype, "open", {
        /**
         * Will determine if acordion is expanded or not
         */
        get: function () {
            return this.isOpen;
        },
        set: function (val) {
            if (val !== this.isOpen) {
                this.isOpen = this.coerceBooleanInput(val);
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * Will toggle open state and emit on Change event
     * @param evt HTML Click Event
     */
    AccordionComponent.prototype.toggleState = function (evt) {
        evt.stopPropagation();
        this.open = !this.open;
        this.change.emit(this.open);
    };
    /**
     * Generate classes using BEM appraoch and bind to ngClass
     */
    AccordionComponent.prototype.generateClassList = function () {
        var _a;
        var base = 'fj-accordion--';
        var classes = (_a = {},
            _a[base + "open"] = this.open,
            _a[base + "disabled"] = this.disabled,
            _a[base + "label-after"] = this.labelPosition === 'after',
            _a);
        return classes;
    };
    AccordionComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-accordion',
                    template: "<div class=\"fj-accordion\"\n     [id]=\"id\"\n     [ngClass]=\"generateClassList()\">\n  <div class=\"fj-accordion__header\"\n       (click)=\"toggleState($event)\">\n    <label>{{label}}</label>\n  </div>\n  <div class=\"fj-accordion__content\">\n    <ng-content></ng-content>\n  </div>\n</div>",
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-accordion label{display:block;font-weight:700;padding-bottom:.6875rem;padding-top:.6875rem;transition:.2s linear;cursor:pointer}.fj-accordion label::before{border-right:1px solid;border-top:1px solid;content:'';display:inline-block;height:6px;margin:0 1rem 0 .25rem;-webkit-transform:rotate(45deg);transform:rotate(45deg);-webkit-transform-origin:80% 40%;transform-origin:80% 40%;transition:.2s linear;width:6px}.fj-accordion__header{display:flex;cursor:pointer}.fj-accordion__content{max-height:0;opacity:0;overflow:hidden;padding:0 1.25rem;transition:.2s linear .1s;border:1px solid rgba(0,0,0,.15)}.fj-accordion--disabled{color:rgba(0,0,0,.25);pointer-events:none}.fj-accordion--open .fj-accordion__content{max-height:15em;opacity:1;overflow:auto;padding:.6875rem 1.25rem}.fj-accordion--open label::before{-webkit-transform:rotate(135deg);transform:rotate(135deg)}.fj-accordion--label-after>div{justify-content:flex-end}"]
                }] }
    ];
    /** @nocollapse */
    AccordionComponent.ctorParameters = function () { return []; };
    AccordionComponent.propDecorators = {
        id: [{ type: Input }],
        open: [{ type: Input }],
        change: [{ type: Output }]
    };
    return AccordionComponent;
}(BaseComponent));
export { AccordionComponent };
