// freyja-ui/esm2015/lib/accordion/accordion.component.js
// Angular Imports
import { Component, Input, Output, EventEmitter } from '@angular/core';
// Internal Imports
import { BaseComponent } from '../shared/base.component';
/**
 * Counter for Next Id
 */
let nextId = 0;
/**
 * Freyja Accordion Component according to specifics of Freyja Design
 */
export class AccordionComponent extends BaseComponent {
    constructor() {
        super();
        // This is to make component open state controlled
        this.isOpen = false;
        /**
         * Provide unique ID or it will use auto generated from counter
         */
        this.id = `fj-accordion-${++nextId}`;
        /**
         * Will emit if open state get's changed on click
         */
        this.change = new EventEmitter(false);
    }
    /**
     * Will determine if acordion is expanded or not
     */
    get open() {
        return this.isOpen;
    }
    set open(val) {
        if (val !== this.isOpen) {
            this.isOpen = this.coerceBooleanInput(val);
        }
    }
    /**
     * Will toggle open state and emit on Change event
     * @param evt HTML Click Event
     */
    toggleState(evt) {
        evt.stopPropagation();
        this.open = !this.open;
        this.change.emit(this.open);
    }
    /**
     * Generate classes using BEM appraoch and bind to ngClass
     */
    generateClassList() {
        const base = 'fj-accordion--';
        const classes = {
            [`${base}open`]: this.open,
            [`${base}disabled`]: this.disabled,
            [`${base}label-after`]: this.labelPosition === 'after'
        };
        return classes;
    }
}
AccordionComponent.decorators = [
    { type: Component, args: [{
                selector: 'fj-accordion',
                template: "<div class=\"fj-accordion\"\n     [id]=\"id\"\n     [ngClass]=\"generateClassList()\">\n  <div class=\"fj-accordion__header\"\n       (click)=\"toggleState($event)\">\n    <label>{{label}}</label>\n  </div>\n  <div class=\"fj-accordion__content\">\n    <ng-content></ng-content>\n  </div>\n</div>",
                styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-accordion label{display:block;font-weight:700;padding-bottom:.6875rem;padding-top:.6875rem;transition:.2s linear;cursor:pointer}.fj-accordion label::before{border-right:1px solid;border-top:1px solid;content:'';display:inline-block;height:6px;margin:0 1rem 0 .25rem;-webkit-transform:rotate(45deg);transform:rotate(45deg);-webkit-transform-origin:80% 40%;transform-origin:80% 40%;transition:.2s linear;width:6px}.fj-accordion__header{display:flex;cursor:pointer}.fj-accordion__content{max-height:0;opacity:0;overflow:hidden;padding:0 1.25rem;transition:.2s linear .1s;border:1px solid rgba(0,0,0,.15)}.fj-accordion--disabled{color:rgba(0,0,0,.25);pointer-events:none}.fj-accordion--open .fj-accordion__content{max-height:15em;opacity:1;overflow:auto;padding:.6875rem 1.25rem}.fj-accordion--open label::before{-webkit-transform:rotate(135deg);transform:rotate(135deg)}.fj-accordion--label-after>div{justify-content:flex-end}"]
            }] }
];
/** @nocollapse */
AccordionComponent.ctorParameters = () => [];
AccordionComponent.propDecorators = {
    id: [{ type: Input }],
    open: [{ type: Input }],
    change: [{ type: Output }]
};
