// freyja-ui/esm5/lib/number-input/number-input.component.js
import * as tslib_1 from "tslib";
// Angular Imports
import { Component, ViewEncapsulation, Input, Output, EventEmitter } from '@angular/core';
// Internal Imports
import { BaseComponent } from '../shared/base.component';
/**
 * Count for Next Id
 */
var nextId = 0;
/**
 * Freyja Number Input Component according to specifics of Freyja Design
 */
var NumberInputComponent = /** @class */ (function (_super) {
    tslib_1.__extends(NumberInputComponent, _super);
    function NumberInputComponent() {
        var _this = _super.call(this) || this;
        // This is to make component controlled
        _this.inputValue = '';
        /**
         * Provide unique ID or it will use auto generated from counter
         */
        _this.id = "fj-number-input-" + ++nextId;
        /**
         * Allow applying forceful error state of input
         */
        _this.errored = false;
        /**
         * On change event hook, which will be emitted on value change
         */
        _this.change = new EventEmitter();
        return _this;
    }
    /**
     * Capture value change ang emit on change event callback
     * @param event HTML onChange Event
     */
    NumberInputComponent.prototype.onValueChange = function (event) {
        event.stopPropagation();
        this.inputValue = event.target.value;
        this.change.emit(this.inputValue);
    };
    /**
     * Generate classes using BEM appraoch and bind to ngClass
     */
    NumberInputComponent.prototype.generateClassList = function () {
        var _a;
        var classes = (_a = {},
            _a['fj-number-input--' + this.size] = this.size,
            _a['fj-number-input--error'] = this.errored,
            _a);
        return classes;
    };
    NumberInputComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-number-input',
                    template: "<label [attr.for]=\"id\" class=\"label--medium\">{{label}}</label>\n<input \n  class=\"fj-input\"\n  [ngClass]=\"generateClassList()\"\n  type=\"number\" \n  [id]=\"id\"\n  [tabIndex]=\"tabIndex\"\n  [disabled]=\"disabled\"\n  [attr.name]=\"name\" \n  [placeholder]=\"placeholder\"\n  [attr.value]=\"value\"\n  (change)=\"onValueChange($event)\"\n/>",
                    encapsulation: ViewEncapsulation.None,
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-input input{box-sizing:border-box;border:.0625rem solid rgba(0,0,0,.15);border-radius:.25rem;color:#000;display:block;font-size:.95rem;font-weight:500;height:2.5rem;margin-top:.25em;outline:0;padding:.5625rem .75rem;transition:.2s linear}.fj-input input:focus{border-color:#1b51a0}.fj-input input:hover{color:rgba(0,0,0,.6)}.fj-input input::-webkit-input-placeholder{color:rgba(0,0,0,.6);font-weight:100}.fj-input input::-ms-input-placeholder{color:rgba(0,0,0,.6);font-weight:100}.fj-input input::placeholder{color:rgba(0,0,0,.6);font-weight:100}.fj-input input::-webkit-input-placeholder:hover .fj-input input::-webkit-input-placeholder{color:rgba(0,0,0,.6)}.fj-input input::-ms-input-placeholder:hover .fj-input input::-ms-input-placeholder{color:rgba(0,0,0,.6)}.fj-input input::placeholder:hover .fj-input input::placeholder{color:rgba(0,0,0,.6)}.fj-input input:disabled{background-color:#f2f2f2;border-color:#f2f2f2;cursor:not-allowed}.fj-input input:disabled::-webkit-input-placeholder{color:rgba(0,0,0,.2)}.fj-input input:disabled::-ms-input-placeholder{color:rgba(0,0,0,.2)}.fj-input input:disabled::placeholder{color:rgba(0,0,0,.2)}.fj-input input--small{height:2rem;font-size:.875rem;padding:.5625rem .625rem}.fj-input input--micro{height:1.5rem;font-size:.75rem;padding:.5625rem .5rem}.fj-input input--required:after{content:\" *\"}.fj-input--error input{border-color:#db0000}label{cursor:pointer;font-size:.875rem}label.label--medium{font-size:.75rem}label.label--disabled{color:rgba(0,0,0,.2)}.input-error{color:#db0000;font-size:.75rem;padding:.5em}.error--label{font-size:.75rem;padding-left:.5em;color:#db0000}"]
                }] }
    ];
    /** @nocollapse */
    NumberInputComponent.ctorParameters = function () { return []; };
    NumberInputComponent.propDecorators = {
        id: [{ type: Input }],
        value: [{ type: Input }],
        errored: [{ type: Input }],
        change: [{ type: Output }]
    };
    return NumberInputComponent;
}(BaseComponent));
export { NumberInputComponent };
