// freyja-ui/esm5/lib/shared/base.component.js
// Angular Imports
import { Input } from '@angular/core';
/**
 * @description Common set of inputs and methods to be used in all Freyja Components
 */
var BaseComponent = /** @class */ (function () {
    function BaseComponent() {
        this._disabled = false;
        /**
         * If component will take full width of it's container
         */
        this.fullWidth = false;
        /**
         * label position, if it will appear before or after content
         */
        this.labelPosition = 'after';
        /**
       * Size can be micro, small, medium or large
       */
        this.size = 'medium';
        /**
         * Placeholder for HTML input elements
         */
        this.placeholder = '';
        /**
         * if component value is required in a form
         */
        this.required = false;
    }
    Object.defineProperty(BaseComponent.prototype, "disabled", {
        /**
         * If component will be disabled, setter/getter to support approach like below
         * <fj-comp disabled></fj-comp>
         */
        get: function () {
            return this._disabled;
        },
        set: function (v) {
            this._disabled = this.coerceBooleanInput(v);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @description coerce an inout value into boolean
     * @param val an input value which will be coearced into a boolean value
     */
    BaseComponent.prototype.coerceBooleanInput = function (val) {
        return !!val || val === '';
    };
    BaseComponent.propDecorators = {
        customStyles: [{ type: Input }],
        name: [{ type: Input }],
        margin: [{ type: Input }],
        disabled: [{ type: Input }],
        fullWidth: [{ type: Input }],
        label: [{ type: Input }],
        labelPosition: [{ type: Input }],
        size: [{ type: Input }],
        placeholder: [{ type: Input }],
        required: [{ type: Input }],
        tabIndex: [{ type: Input }],
        ariaLabel: [{ type: Input, args: ['aria-label',] }],
        ariaLabelledby: [{ type: Input, args: ['aria-labelledby',] }],
        ariaDescribedby: [{ type: Input, args: ['aria-describedby',] }]
    };
    return BaseComponent;
}());
export { BaseComponent };
