// freyja-ui/esm5/lib/button/button.component.js
import * as tslib_1 from "tslib";
// Angular imports
import { Component, Input } from '@angular/core';
// Internal imports
import { IconBaseComponent } from '../shared/icon-base.component';
/**
 * Freyja Button Component using Freyja Design specifics
 * it extends properties of icon base component
 */
var nextId = 0;
var ButtonComponent = /** @class */ (function (_super) {
    tslib_1.__extends(ButtonComponent, _super);
    // TODO: add generic generateClass methods
    function ButtonComponent() {
        var _this = _super.call(this) || this;
        /** Determin if button is ghost or not, and add relevant css classes */
        _this.ghost = false;
        /**
         * Button type, it can be button, submit or reset
         */
        _this.type = 'button';
        /**
         * Will determin look and feel of button
         */
        _this.color = 'primary';
        /**
         * Shape of button, it can be rounder, square or radius
         */
        _this.shape = 'radius';
        /**
         * Provide unique to be used for button
         */
        _this.id = "fj-button-" + ++nextId;
        return _this;
    }
    /**
     * @description Perform operations which needs to be done after component is initialized
     */
    ButtonComponent.prototype.ngOnInit = function () {
        if (this.ghost) {
            this.iconFill = 'blue';
        }
    };
    /**
     * @description Add css classes based on inputs like size, shape etc
     */
    ButtonComponent.prototype.generateClassList = function () {
        var _a;
        var baseCls = 'fj-button--';
        var classes = (_a = {},
            _a[baseCls + this.color] = this.color,
            _a[baseCls + this.size] = this.size,
            _a[baseCls + this.shape] = this.shape,
            _a[baseCls + 'full-width'] = this.fullWidth,
            _a[baseCls + "icon-before"] = this.iconPosition === 'before',
            _a[baseCls + "ghost"] = this.ghost,
            _a);
        return classes;
    };
    /**
     * generate custom styles
     */
    ButtonComponent.prototype.generateStyles = function () {
        return tslib_1.__assign({}, this.customStyles, { margin: this.margin });
    };
    /**
     * function that it is triggered when the mouse pointer enters the button
     */
    ButtonComponent.prototype.onMouseEnter = function () {
        if (this.ghost) {
            this.iconFill = 'white';
        }
    };
    /**
     * function that it is triggered when the mouse pointer leaves the button
     */
    ButtonComponent.prototype.onMouseLeave = function () {
        if (this.ghost) {
            this.iconFill = 'blue';
        }
    };
    ButtonComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-button',
                    template: "<button [id]=\"id\"\n        class=\"fj-button\"\n        [ngClass]=\"generateClassList()\"\n        [ngStyle]= \"generateStyles()\"\n        [disabled]=\"disabled\"\n        [tabIndex]=\"tabIndex\"\n        [type]=\"type\"\n        (mouseenter)=\"onMouseEnter()\"\n        (mouseleave)=\"onMouseLeave()\"\n>\n  <span class=\"fj-button-label\">\n    <ng-content></ng-content>\n  </span>\n  <fj-icon *ngIf=\"icon\"\n           class=\"fj-button--icon\"\n           [ngStyle]=\"{margin: iconMargin}\"\n           [svgName]=\"icon\"\n           [size]=\"size\"\n           [fillColor]=\"iconFill\">\n  </fj-icon>\n</button>\n",
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-button{display:inline-flex;color:#fff;white-space:nowrap;background-color:#1b51a0;border:.0625rem solid transparent;cursor:pointer;transition:.2s linear}.fj-button:active,.fj-button:focus,.fj-button:hover{background-color:#7596c5}.fj-button--rounded{border-radius:2rem}.fj-button--radius{border-radius:.25rem}.fj-button--light{background-color:transparent;border-color:#1b51a0;color:#1b51a0}.fj-button--light:active,.fj-button--light:focus,.fj-button--light:hover{background-color:#dce8f9}.fj-button--light.btn--danger{background-color:transparent;border-color:#db0000;color:#db0000}.fj-button--light.btn--danger:active,.fj-button--light.btn--danger:focus,.fj-button--light.btn--danger:hover{background-color:#ffe9e9}.fj-button--light:disabled{border-color:rgba(0,0,0,.15);background-color:transparent;color:rgba(0,0,0,.15)}.fj-button--ghost.fj-button{background-color:transparent;color:#1b51a0;transition:.1s linear}.fj-button--ghost.fj-button:active,.fj-button--ghost.fj-button:focus,.fj-button--ghost.fj-button:hover{background-color:#1b51a0;color:#fff}.fj-button--ghost.fj-button:disabled{background-color:transparent;color:rgba(0,0,0,.15)}.fj-button--ghost.fj-button:disabled:disabled::after,.fj-button--ghost.fj-button:disabled:disabled::before{border-color:rgba(0,0,0,.15)}.fj-button--ghost.fj-button.btn--danger{background-color:transparent;color:#db0000}.fj-button--ghost.fj-button.btn--danger:active,.fj-button--ghost.fj-button.btn--danger:focus,.fj-button--ghost.fj-button.btn--danger:hover{background-color:#db0000;color:#fff}.fj-button--ghost.fj-button.btn--danger span{background-color:#db0000}.fj-button--ghost.fj-button.btn--light{border-color:transparent}.fj-button--ghost.fj-button.btn--light::after,.fj-button--ghost.fj-button.btn--light::before{border-color:#1b51a0}.fj-button--ghost.fj-button.btn--light:active::after,.fj-button--ghost.fj-button.btn--light:active::before,.fj-button--ghost.fj-button.btn--light:focus::after,.fj-button--ghost.fj-button.btn--light:focus::before,.fj-button--ghost.fj-button.btn--light:hover::after,.fj-button--ghost.fj-button.btn--light:hover::before{border-color:#fff}.fj-button--ghost.fj-button.btn--light span{background-color:transparent}.fj-button--ghost.fj-button.btn--light.btn--danger::after,.fj-button--ghost.fj-button.btn--light.btn--danger::before{border-color:#db0000}.fj-button--ghost.fj-button.btn--light.btn--danger:active::after,.fj-button--ghost.fj-button.btn--light.btn--danger:active::before,.fj-button--ghost.fj-button.btn--light.btn--danger:focus::after,.fj-button--ghost.fj-button.btn--light.btn--danger:focus::before,.fj-button--ghost.fj-button.btn--light.btn--danger:hover::after,.fj-button--ghost.fj-button.btn--light.btn--danger:hover::before{border-color:#fff}.fj-button--full-width{width:100%;display:flex;justify-content:center}.fj-button--large{font-size:1rem;height:3rem;line-height:1.25rem;padding:.875rem 1.25rem}.fj-button--medium{font-size:.875rem;height:2.5rem;line-height:1.125rem;padding:.6875rem 1rem}.fj-button--small{font-size:.75rem;height:2rem;line-height:1rem;padding:.5rem .75rem}.fj-button--micro{font-size:.625rem;height:1.5rem;line-height:.75rem;padding:.375rem .5rem}.fj-button--danger{background-color:#db0000}.fj-button--danger:active,.fj-button--danger:focus,.fj-button--danger:hover{background-color:#e96565}.fj-button:disabled{background-color:rgba(0,0,0,.15);pointer-events:none}.fj-button fj-icon{margin-left:1rem}.fj-button--icon-before>span{order:1}.fj-button--icon-before fj-icon{margin-right:1rem;margin-left:0}"]
                }] }
    ];
    /** @nocollapse */
    ButtonComponent.ctorParameters = function () { return []; };
    ButtonComponent.propDecorators = {
        ghost: [{ type: Input }],
        type: [{ type: Input }],
        color: [{ type: Input }],
        shape: [{ type: Input }],
        id: [{ type: Input }]
    };
    return ButtonComponent;
}(IconBaseComponent));
export { ButtonComponent };
