// freyja-ui/esm2015/lib/button/button.component.js
// Angular imports
import { Component, Input } from '@angular/core';
// Internal imports
import { IconBaseComponent } from '../shared/icon-base.component';
/**
 * Freyja Button Component using Freyja Design specifics
 * it extends properties of icon base component
 */
let nextId = 0;
export class ButtonComponent extends IconBaseComponent {
    // TODO: add generic generateClass methods
    constructor() {
        super();
        /** Determin if button is ghost or not, and add relevant css classes */
        this.ghost = false;
        /**
         * Button type, it can be button, submit or reset
         */
        this.type = 'button';
        /**
         * Will determin look and feel of button
         */
        this.color = 'primary';
        /**
         * Shape of button, it can be rounder, square or radius
         */
        this.shape = 'radius';
        /**
         * Provide unique to be used for button
         */
        this.id = `fj-button-${++nextId}`;
    }
    /**
     * @description Perform operations which needs to be done after component is initialized
     */
    ngOnInit() {
        if (this.ghost) {
            this.iconFill = 'blue';
        }
    }
    /**
     * @description Add css classes based on inputs like size, shape etc
     */
    generateClassList() {
        const baseCls = 'fj-button--';
        const classes = {
            [baseCls + this.color]: this.color,
            [baseCls + this.size]: this.size,
            [baseCls + this.shape]: this.shape,
            [baseCls + 'full-width']: this.fullWidth,
            [`${baseCls}icon-before`]: this.iconPosition === 'before',
            [`${baseCls}ghost`]: this.ghost
        };
        return classes;
    }
    /**
     * generate custom styles
     */
    generateStyles() {
        return Object.assign({}, this.customStyles, { margin: this.margin });
    }
    /**
     * function that it is triggered when the mouse pointer enters the button
     */
    onMouseEnter() {
        if (this.ghost) {
            this.iconFill = 'white';
        }
    }
    /**
     * function that it is triggered when the mouse pointer leaves the button
     */
    onMouseLeave() {
        if (this.ghost) {
            this.iconFill = 'blue';
        }
    }
}
ButtonComponent.decorators = [
    { type: Component, args: [{
                selector: 'fj-button',
                template: "<button [id]=\"id\"\n        class=\"fj-button\"\n        [ngClass]=\"generateClassList()\"\n        [ngStyle]= \"generateStyles()\"\n        [disabled]=\"disabled\"\n        [tabIndex]=\"tabIndex\"\n        [type]=\"type\"\n        (mouseenter)=\"onMouseEnter()\"\n        (mouseleave)=\"onMouseLeave()\"\n>\n  <span class=\"fj-button-label\">\n    <ng-content></ng-content>\n  </span>\n  <fj-icon *ngIf=\"icon\"\n           class=\"fj-button--icon\"\n           [ngStyle]=\"{margin: iconMargin}\"\n           [svgName]=\"icon\"\n           [size]=\"size\"\n           [fillColor]=\"iconFill\">\n  </fj-icon>\n</button>\n",
                styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-button{display:inline-flex;color:#fff;white-space:nowrap;background-color:#1b51a0;border:.0625rem solid transparent;cursor:pointer;transition:.2s linear}.fj-button:active,.fj-button:focus,.fj-button:hover{background-color:#7596c5}.fj-button--rounded{border-radius:2rem}.fj-button--radius{border-radius:.25rem}.fj-button--light{background-color:transparent;border-color:#1b51a0;color:#1b51a0}.fj-button--light:active,.fj-button--light:focus,.fj-button--light:hover{background-color:#dce8f9}.fj-button--light.btn--danger{background-color:transparent;border-color:#db0000;color:#db0000}.fj-button--light.btn--danger:active,.fj-button--light.btn--danger:focus,.fj-button--light.btn--danger:hover{background-color:#ffe9e9}.fj-button--light:disabled{border-color:rgba(0,0,0,.15);background-color:transparent;color:rgba(0,0,0,.15)}.fj-button--ghost.fj-button{background-color:transparent;color:#1b51a0;transition:.1s linear}.fj-button--ghost.fj-button:active,.fj-button--ghost.fj-button:focus,.fj-button--ghost.fj-button:hover{background-color:#1b51a0;color:#fff}.fj-button--ghost.fj-button:disabled{background-color:transparent;color:rgba(0,0,0,.15)}.fj-button--ghost.fj-button:disabled:disabled::after,.fj-button--ghost.fj-button:disabled:disabled::before{border-color:rgba(0,0,0,.15)}.fj-button--ghost.fj-button.btn--danger{background-color:transparent;color:#db0000}.fj-button--ghost.fj-button.btn--danger:active,.fj-button--ghost.fj-button.btn--danger:focus,.fj-button--ghost.fj-button.btn--danger:hover{background-color:#db0000;color:#fff}.fj-button--ghost.fj-button.btn--danger span{background-color:#db0000}.fj-button--ghost.fj-button.btn--light{border-color:transparent}.fj-button--ghost.fj-button.btn--light::after,.fj-button--ghost.fj-button.btn--light::before{border-color:#1b51a0}.fj-button--ghost.fj-button.btn--light:active::after,.fj-button--ghost.fj-button.btn--light:active::before,.fj-button--ghost.fj-button.btn--light:focus::after,.fj-button--ghost.fj-button.btn--light:focus::before,.fj-button--ghost.fj-button.btn--light:hover::after,.fj-button--ghost.fj-button.btn--light:hover::before{border-color:#fff}.fj-button--ghost.fj-button.btn--light span{background-color:transparent}.fj-button--ghost.fj-button.btn--light.btn--danger::after,.fj-button--ghost.fj-button.btn--light.btn--danger::before{border-color:#db0000}.fj-button--ghost.fj-button.btn--light.btn--danger:active::after,.fj-button--ghost.fj-button.btn--light.btn--danger:active::before,.fj-button--ghost.fj-button.btn--light.btn--danger:focus::after,.fj-button--ghost.fj-button.btn--light.btn--danger:focus::before,.fj-button--ghost.fj-button.btn--light.btn--danger:hover::after,.fj-button--ghost.fj-button.btn--light.btn--danger:hover::before{border-color:#fff}.fj-button--full-width{width:100%;display:flex;justify-content:center}.fj-button--large{font-size:1rem;height:3rem;line-height:1.25rem;padding:.875rem 1.25rem}.fj-button--medium{font-size:.875rem;height:2.5rem;line-height:1.125rem;padding:.6875rem 1rem}.fj-button--small{font-size:.75rem;height:2rem;line-height:1rem;padding:.5rem .75rem}.fj-button--micro{font-size:.625rem;height:1.5rem;line-height:.75rem;padding:.375rem .5rem}.fj-button--danger{background-color:#db0000}.fj-button--danger:active,.fj-button--danger:focus,.fj-button--danger:hover{background-color:#e96565}.fj-button:disabled{background-color:rgba(0,0,0,.15);pointer-events:none}.fj-button fj-icon{margin-left:1rem}.fj-button--icon-before>span{order:1}.fj-button--icon-before fj-icon{margin-right:1rem;margin-left:0}"]
            }] }
];
/** @nocollapse */
ButtonComponent.ctorParameters = () => [];
ButtonComponent.propDecorators = {
    ghost: [{ type: Input }],
    type: [{ type: Input }],
    color: [{ type: Input }],
    shape: [{ type: Input }],
    id: [{ type: Input }]
};
