// freyja-ui/esm5/lib/modal/modal.component.js
import { Component, EventEmitter, Input, Output, ElementRef } from '@angular/core';
var ModalComponent = /** @class */ (function () {
    function ModalComponent(eRef) {
        this.eRef = eRef;
        /**
         * Content of the modal window
         */
        this.modalContent = {};
        /**
         * an output that will be emitted on an action is triggered in the modal
         * @type {EventEmitter<any>}
         */
        this.actionClicked = new EventEmitter();
        /**
         * flag to show or hide the modal
         * @type {boolean}
         */
        this.showModal = false;
    }
    ModalComponent.prototype.ngOnInit = function () {
        if (!this.size) {
            this.size = 'small';
        }
    };
    /**
     * fired when an action has been clicked
     * @param evt - click event
     * @param actionType - type of the action that has been fired
     */
    ModalComponent.prototype.onActionClicked = function (evt, actionType) {
        if (evt.stopPropagation) {
            evt.stopPropagation();
        }
        else {
            actionType = evt;
        }
        this.actionClicked.emit(actionType);
        this.showModal = false;
    };
    /**
     * this function avoid closing the modal when clicking on its content
     * @param evt
     */
    ModalComponent.prototype.modalClicked = function (evt) {
        evt.stopPropagation();
    };
    ModalComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-modal',
                    template: "<div class=\"modal\" [ngClass]=\"{'modal--hidden': !showModal}\" (click)=\"showModal = false\">\n  <!-- Modal content -->\n  <div class=\"modal__content\" (click)=\"modalClicked($event)\" [ngClass]=\"{'modal__content--big': size === 'big', 'modal__content--small': size === 'small'}\" [ngStyle]=\"customStyles\" >\n    <span class=\"modal__content__close\" (click)=\"onActionClicked($event, 'close')\">&times;</span>\n    <div class=\"modal__content__top\">\n      <h3 *ngIf=\"modalContent.title\" class=\"modal__content__top__title\">{{modalContent.title}}</h3>\n    </div>\n    <div class=\"modal__content__body\">\n      <p *ngIf=\"modalContent.text\">{{modalContent.text}}</p>\n      <ng-content select=\"[modal-content]\"></ng-content>\n    </div>\n    <div class=\"modal__content__bottom\">\n      <fj-button class=\"modal__content__button\" *ngIf=\"modalContent.primaryAction\" (click)=\"onActionClicked($event, 'primary')\">{{modalContent.primaryAction.label}}</fj-button>\n      <fj-button class=\"modal__content__button\" *ngIf=\"modalContent.dangerAction\" (click)=\"onActionClicked($event, 'danger')\" [color]=\"'danger'\">{{modalContent.dangerAction.label}}</fj-button>\n      <fj-button class=\"modal__content__button\" *ngIf=\"modalContent.secondaryAction\" (click)=\"onActionClicked($event, 'secondary')\" [color]=\"'light'\">{{modalContent.secondaryAction.label}}</fj-button>\n    </div>\n  </div>\n</div>\n",
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.modal{position:fixed;z-index:999;left:0;top:0;width:100%;height:100%;background-color:rgba(0,0,0,.4)}.modal--hidden{display:none}.modal__content{position:relative;display:flex;flex-direction:column;background-color:#fff;margin:10% auto;padding:1.5625rem;border:1px solid #888;box-shadow:0 1px 4px 0 rgba(0,0,0,.3)}.modal__content--big{max-width:60%}.modal__content--small{max-width:40%}.modal__content__top{display:flex;align-items:center}.modal__content__top__title{flex:80%;margin:0}.modal__content__body{display:flex;flex-wrap:wrap}.modal__content__bottom{display:flex;flex-direction:row-reverse;flex-wrap:wrap}.modal__content__close{position:absolute;right:.9375rem;top:.4375rem;color:rgba(0,0,0,.45);font-weight:lighter;font-size:20px}.modal__content__close:focus,.modal__content__close:hover{color:#000;text-decoration:none;cursor:pointer}.modal__content__button{margin-right:.3125rem}"]
                }] }
    ];
    /** @nocollapse */
    ModalComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    ModalComponent.propDecorators = {
        modalContent: [{ type: Input }],
        size: [{ type: Input }],
        customStyles: [{ type: Input }],
        actionClicked: [{ type: Output }]
    };
    return ModalComponent;
}());
export { ModalComponent };
