// freyja-ui/esm2015/lib/checkbox/checkbox.component.js
// Angular Imports
import { Component, Input, Output, EventEmitter } from '@angular/core';
// Internal Imports
import { BaseComponent } from '../shared/base.component';
/**
 * Count for Next Id
 */
let nextId = 0;
/**
 * Freyja CheckBox Component according to specifics of Freyja Design
 */
export class CheckboxComponent extends BaseComponent {
    constructor() {
        super();
        // This is to make component controlled
        this.checkedVal = false;
        this.indeterminateVal = false;
        /**
         * Provide unique ID or it will use auto generated from counter
         */
        this.id = `fj-checkbox-${++nextId}`;
        /**
         * on change event, will be emitted once user clicks on checkbox to change value
         */
        this.change = new EventEmitter(false);
    }
    /**
     * will determine indeterminate state of checkbox
     */
    get indeterminate() {
        return this.indeterminateVal;
    }
    set indeterminate(val) {
        this.indeterminateVal = this.coerceBooleanInput(val);
    }
    /**
     * Will determin initial checked state of checkbox
     */
    get checked() {
        return this.checkedVal;
    }
    set checked(v) {
        if (v !== this.checkedVal) {
            this.checkedVal = this.coerceBooleanInput(v);
        }
    }
    /**
     * capture on click and toggle state of checkbox
     * @param event emitted by html element on click
     */
    onCheckboxClicked(event) {
        event.stopPropagation();
        this.toggle();
    }
    /**
     * stop event propagation by capturing it
     * @param event html event
     */
    onChangeEvent(event) {
        event.stopPropagation();
    }
    /**
     * Generate classes using BEM appraoch and bind to ngClass
     */
    generateClassList() {
        const base = 'fj-checkbox--';
        const classes = {
            [`${base}label-before`]: this.labelPosition === 'before',
            [`${base}disabled`]: this.disabled,
            [`${base}indeterminate`]: this.indeterminate
        };
        return classes;
    }
    /**
     * Toggle state of checked value and emit on change
     */
    toggle() {
        this.checked = !this.checked;
        this.change.emit(this.checkedVal);
    }
}
CheckboxComponent.decorators = [
    { type: Component, args: [{
                selector: 'fj-checkbox',
                template: "<label [attr.for]=\"id\"\n       class=\"fj-checkbox\"\n       [ngClass]=\"generateClassList()\">\n  <input class=\"fj-checkbox__input\"\n         type=\"checkbox\"\n         [id]=\"id\"\n         [required]=\"required\"\n         [checked]=\"checked\"\n         [disabled]=\"disabled\"\n         [attr.name]=\"name\"\n         [tabIndex]=\"tabIndex\"\n         [indeterminate]=\"indeterminate\"\n         [attr.aria-label]=\"ariaLabel || null\"\n         [attr.aria-labelledby]=\"ariaLabelledby\"\n         (change)=\"onChangeEvent($event)\"\n         (click)=\"onCheckboxClicked($event)\">\n  <span class=\"fj-checkbox-label\">\n    <ng-content></ng-content>\n  </span>\n</label>",
                styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-checkbox{display:inline-flex;cursor:pointer;white-space:nowrap;align-items:center;margin-right:.5625rem}.fj-checkbox__input{display:inline-block;-webkit-appearance:none;-moz-appearance:none;appearance:none;cursor:pointer;height:1.125rem;outline:0;position:relative;width:1.125rem;margin-right:.5625rem}.fj-checkbox__input::before{display:block;position:absolute;top:0;left:0;background-color:transparent;border:.0625rem solid #1b51a0;border-radius:.125rem;content:'';height:1rem;transition:.1s linear;width:1rem}.fj-checkbox__input::after{border-bottom:.125rem solid #fff;border-radius:.0625rem;border-right:.125rem solid #fff;content:'';display:block;height:.5625rem;left:.375rem;opacity:0;position:absolute;top:.125rem;-webkit-transform:rotate(45deg);transform:rotate(45deg);width:.25rem}.fj-checkbox__input:active::before,.fj-checkbox__input:focus::before,.fj-checkbox__input:hover::before{background-color:#dce8f9}.fj-checkbox__input:checked::before{background-color:#1b51a0}.fj-checkbox__input:checked::after{opacity:1}.fj-checkbox__input:checked:active::before,.fj-checkbox__input:checked:focus::before,.fj-checkbox__input:checked:hover::before{background-color:#7596c5}.fj-checkbox__input:disabled{opacity:.4;cursor:default}.fj-checkbox__input:disabled::after{background-color:transparent;border-color:#000}.fj-checkbox__input:disabled:checked::before{background-color:rgba(0,0,0,.15)}.fj-checkbox__input:indeterminate::after{-webkit-transform:rotate(45deg);transform:rotate(45deg)}.fj-checkbox--label-before .fj-checkbox-input{order:1;margin-left:.5625rem}.fj-checkbox--disabled{opacity:.5;cursor:auto}"]
            }] }
];
/** @nocollapse */
CheckboxComponent.ctorParameters = () => [];
CheckboxComponent.propDecorators = {
    id: [{ type: Input }],
    indeterminate: [{ type: Input }],
    change: [{ type: Output }],
    checked: [{ type: Input }]
};
