// freyja-ui/esm2015/lib/drag-and-drop/drag-and-drop.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropzoneModule } from 'ngx-dropzone-wrapper';
import { DROPZONE_CONFIG } from 'ngx-dropzone-wrapper';
import { FjDragAndDropComponent } from './drag-and-drop.component';
import { IconModule } from '../icon/icon.module';
import { FjButtonModule } from '../button/button.module';
const DEFAULT_DROPZONE_CONFIGURATIONS = {
    url: '/',
    maxFiles: 1,
    dictDefaultMessage: '',
    createImageThumbnails: true,
    acceptedFiles: 'image/*, .pdf, .doc, .docx'
};
const ɵ0 = DEFAULT_DROPZONE_CONFIGURATIONS;
export class FjDragAndDropModule {
}
FjDragAndDropModule.decorators = [
    { type: NgModule, args: [{
                declarations: [FjDragAndDropComponent],
                imports: [
                    CommonModule,
                    DropzoneModule,
                    FjButtonModule,
                    IconModule
                ],
                exports: [FjDragAndDropComponent, DropzoneModule],
                providers: [
                    {
                        provide: DROPZONE_CONFIG,
                        useValue: ɵ0
                    }
                ]
            },] }
];
export { ɵ0 };
