// freyja-ui/esm5/lib/text-input/text-input.component.js
import * as tslib_1 from "tslib";
// Angular Imports
import { Component, Input, Output, EventEmitter } from '@angular/core';
// Internal Imports
import { BaseComponent } from '../shared/base.component';
import { FormGroup } from '@angular/forms';
/**
 * Count for Next Id
 */
var nextId = 0;
/**
 * Freyja Text Input Component according to specifics of Freyja Design
 */
var TextInputComponent = /** @class */ (function (_super) {
    tslib_1.__extends(TextInputComponent, _super);
    function TextInputComponent() {
        var _this = _super.call(this) || this;
        /**
         * Represent Internal state of component
         */
        _this.isErrored = false;
        // This is to make component controlled
        _this.inputValue = '';
        /**
         * Unique ID for HTMl Element identifier
         */
        _this.id = "fj-text-input-" + ++nextId;
        /**
         * type of the input, default type is text
         */
        _this.type = 'text';
        /**
         * To Apply forceful error state to input component
         */
        _this.errored = false;
        /**
         * checkes if the element should be focused by default
         */
        _this.focused = false;
        /**
         * Callback for value change
         */
        _this.change = new EventEmitter();
        /**
         * Callback for focus event
         */
        _this.focus = new EventEmitter();
        /**
         * Callback for click event
         */
        _this.click = new EventEmitter();
        /**
         * Callback for mouseout event
         */
        _this.mouseout = new EventEmitter();
        return _this;
    }
    TextInputComponent.prototype.ngOnChanges = function () {
        if (this.focused) {
            document.getElementById(this.id).focus();
        }
    };
    /**
      * Capture value change ang emit on change event callback
      * @param event HTML onChange Event
      */
    TextInputComponent.prototype.onValueChange = function (event) {
        event.preventDefault();
        event.stopImmediatePropagation();
        this.inputValue = event.target.value;
        this.change.emit(this.inputValue);
    };
    /**
     * triggers on focus event
     * @param event HTML onFocus Event
     */
    TextInputComponent.prototype.onFocus = function (event) {
        event.stopPropagation();
        this.focus.emit(this.inputValue);
    };
    /**
     * triggers on click event
     * @param event HTML click Event
     */
    TextInputComponent.prototype.onClick = function (event) {
        event.stopPropagation();
        this.click.emit(this.inputValue);
    };
    /**
     * Capture value change ang emit on mouseout event callback
     * @param event HTML onChange Event
     */
    TextInputComponent.prototype.onMouseOut = function (event) {
        event.stopPropagation();
        this.inputValue = event.target.value;
        this.mouseout.emit(this.inputValue);
    };
    /**
     * Generate classes using BEM appraoch and bind to ngClass
     */
    TextInputComponent.prototype.generateClassList = function () {
        var _a;
        var classes = (_a = {},
            _a['fj-input--' + this.size] = this.size,
            _a['fj-input--required'] = this.required,
            _a['fj-input--error'] = this.errored || this.error,
            _a);
        return classes;
    };
    TextInputComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-text-input',
                    template: "<div class=\"fj-input\" [ngClass]=\"generateClassList()\" [formGroup]=\"formGroup\" *ngIf=\"formGroup\">\n  <label\n    [attr.for]=\"id\"\n    class=\"label--medium\"\n    *ngIf=\"label\"\n    >{{label}}</label>\n  <input\n    type=\"{{type}}\"\n    [ngStyle]=\"customStyles\"\n    [id]=\"id\"\n    [tabIndex]=\"tabIndex\"\n    [disabled]=\"disabled\"\n    [attr.name]=\"name\"\n    [placeholder]=\"placeholder\"\n    formControlName=\"{{controlName}}\"\n    [attr.value]=\"value\"\n    (click)=\"onClick($event)\"\n    (change)=\"onValueChange($event)\"\n    (focus)=\"onFocus($event)\"\n    (mouseout)=\"onMouseOut($event)\"\n  />\n  <label class=\"error--label\" *ngIf=\"error\">{{error}}</label>\n</div>\n<!-- Input without reactive forms-->\n<div class=\"fj-input\" [ngClass]=\"generateClassList()\" *ngIf=\"!formGroup\">\n  <label\n    [attr.for]=\"id\"\n    class=\"label--medium\"\n    *ngIf=\"label\"\n  >{{label}}</label>\n  <input\n    type=\"{{type}}\"\n    [ngStyle]= \"customStyles\"\n    [id]=\"id\"\n    [tabIndex]=\"tabIndex\"\n    [disabled]=\"disabled\"\n    [attr.name]=\"name\"\n    [placeholder]=\"placeholder\"\n    [attr.value]=\"value\"\n    (click)=\"onClick($event)\"\n    (change)=\"onValueChange($event)\"\n    (focus)=\"onFocus($event)\"\n    (mouseout)=\"onMouseOut($event)\"\n  />\n  <label class=\"error--label\" *ngIf=\"error\">{{error}}</label>\n</div>\n",
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-input input{box-sizing:border-box;border:.0625rem solid rgba(0,0,0,.15);border-radius:.25rem;color:#000;display:block;font-size:.95rem;font-weight:500;height:2.5rem;margin-top:.25em;outline:0;padding:.5625rem .75rem;transition:.2s linear}.fj-input input:focus{border-color:#1b51a0}.fj-input input:hover{color:rgba(0,0,0,.6)}.fj-input input::-webkit-input-placeholder{color:rgba(0,0,0,.6);font-weight:100}.fj-input input::-ms-input-placeholder{color:rgba(0,0,0,.6);font-weight:100}.fj-input input::placeholder{color:rgba(0,0,0,.6);font-weight:100}.fj-input input::-webkit-input-placeholder:hover .fj-input input::-webkit-input-placeholder{color:rgba(0,0,0,.6)}.fj-input input::-ms-input-placeholder:hover .fj-input input::-ms-input-placeholder{color:rgba(0,0,0,.6)}.fj-input input::placeholder:hover .fj-input input::placeholder{color:rgba(0,0,0,.6)}.fj-input input:disabled{background-color:#f2f2f2;border-color:#f2f2f2;cursor:not-allowed}.fj-input input:disabled::-webkit-input-placeholder{color:rgba(0,0,0,.2)}.fj-input input:disabled::-ms-input-placeholder{color:rgba(0,0,0,.2)}.fj-input input:disabled::placeholder{color:rgba(0,0,0,.2)}.fj-input input--small{height:2rem;font-size:.875rem;padding:.5625rem .625rem}.fj-input input--micro{height:1.5rem;font-size:.75rem;padding:.5625rem .5rem}.fj-input input--required:after{content:\" *\"}.fj-input--error input{border-color:#db0000}label{cursor:pointer;font-size:.875rem}label.label--medium{font-size:.75rem}label.label--disabled{color:rgba(0,0,0,.2)}.input-error{color:#db0000;font-size:.75rem;padding:.5em}.error--label{font-size:.75rem;padding-left:.5em;color:#db0000}"]
                }] }
    ];
    /** @nocollapse */
    TextInputComponent.ctorParameters = function () { return []; };
    TextInputComponent.propDecorators = {
        id: [{ type: Input }],
        customStyles: [{ type: Input }],
        value: [{ type: Input }],
        type: [{ type: Input }],
        errored: [{ type: Input }],
        focused: [{ type: Input }],
        error: [{ type: Input }],
        change: [{ type: Output }],
        focus: [{ type: Output }],
        click: [{ type: Output }],
        mouseout: [{ type: Output }],
        controlName: [{ type: Input }],
        formGroup: [{ type: Input }]
    };
    return TextInputComponent;
}(BaseComponent));
export { TextInputComponent };
