// freyja-ui/esm5/lib/drag-and-drop/drag-and-drop.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropzoneModule } from 'ngx-dropzone-wrapper';
import { DROPZONE_CONFIG } from 'ngx-dropzone-wrapper';
import { FjDragAndDropComponent } from './drag-and-drop.component';
import { IconModule } from '../icon/icon.module';
import { FjButtonModule } from '../button/button.module';
var DEFAULT_DROPZONE_CONFIGURATIONS = {
    url: '/',
    maxFiles: 1,
    dictDefaultMessage: '',
    createImageThumbnails: true,
    acceptedFiles: 'image/*, .pdf, .doc, .docx'
};
var ɵ0 = DEFAULT_DROPZONE_CONFIGURATIONS;
var FjDragAndDropModule = /** @class */ (function () {
    function FjDragAndDropModule() {
    }
    FjDragAndDropModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [FjDragAndDropComponent],
                    imports: [
                        CommonModule,
                        DropzoneModule,
                        FjButtonModule,
                        IconModule
                    ],
                    exports: [FjDragAndDropComponent, DropzoneModule],
                    providers: [
                        {
                            provide: DROPZONE_CONFIG,
                            useValue: ɵ0
                        }
                    ]
                },] }
    ];
    return FjDragAndDropModule;
}());
export { FjDragAndDropModule };
export { ɵ0 };
