// freyja-ui/esm5/lib/drag-and-drop/drag-and-drop.component.js
import { Component, ViewChild, ViewEncapsulation, EventEmitter, Output, Input } from '@angular/core';
import { DropzoneComponent, DropzoneDirective } from 'ngx-dropzone-wrapper';
var FjDragAndDropComponent = /** @class */ (function () {
    function FjDragAndDropComponent() {
        this.uploadedDoc = new EventEmitter();
        this.fileToUpload = new EventEmitter();
        this.disabled = false;
        this.showPlaceholder = true;
        this.dropZoneconfig = {
            clickable: '.trigger-upload',
            maxFiles: 1,
            autoReset: null,
            errorReset: null,
            cancelReset: null,
            uploadMultiple: false
        };
    }
    FjDragAndDropComponent.prototype.ngOnInit = function () { };
    FjDragAndDropComponent.prototype.ngOnChanges = function (changes) {
        this.getParentChanges(changes);
    };
    /**
     * @description called on upload error
     * @param args
     */
    FjDragAndDropComponent.prototype.onUploadError = function (args) {
        this.disabled = true;
        return;
    };
    /**
     * @description on drag-start event
     * @param event
     */
    FjDragAndDropComponent.prototype.onFileAdded = function (file) {
        var _this = this;
        this.uploadedFile = file;
        var reader = new FileReader();
        reader.onload = function (event) {
            // event.target.result contains base64 encoded file
            _this.fileToUpload.emit(reader.result);
        };
        try {
            reader.readAsDataURL(file);
        }
        catch (error) { }
        this.uploadedDoc.emit(file);
        this.showPlaceholder = false;
    };
    /**
     * @description - called on file removed from drag-drop area
     * @description - emits null to the parent
     * @param event
     */
    FjDragAndDropComponent.prototype.onFileRemoved = function (event) {
        this.fileToUpload.emit(null);
        this.showPlaceholder = true;
        this.disabled = false;
    };
    /**
     * @description - resets drag-drop area on form submit and modal close events
     * @description - sets showPlaceholder value to (true/false) to position the drop container
     * @param {object} changes - object of type SimpleChanges
     */
    FjDragAndDropComponent.prototype.getParentChanges = function (changes) {
        if (changes.removeFile && changes.removeFile.currentValue) {
            // clear dropzone
            if (this.directiveRef) {
                this.directiveRef.reset();
            }
            this.showPlaceholder = true;
        }
        // Diplay error message if a form is submitted without selecting a file
        if (changes.noFileSelected &&
            changes.noFileSelected.currentValue === true) {
            this.noFileSelected = true;
        }
        else {
            this.noFileSelected = false;
        }
    };
    FjDragAndDropComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-drag-and-drop',
                    template: "<div class=\"dropzone drag-drop\">\n    <div *ngIf=\"noFileSelected && showPlaceholder\">\n        <p class=\"drag-drop__file-missing\">No file is selected</p>\n    </div>\n    <div\n        class=\"dropzone-container trigger-upload\"\n        [dropzone]=\"dropZoneconfig\"\n        [disabled]=\"disabled\"\n        (error)=\"onUploadError($event)\"\n        (addedFile)=\"onFileAdded($event)\"\n        (removedFile)=\"onFileRemoved($event)\"\n    >\n        <div class=\"dropzone-container__placeholder trigger-upload\">\n            <i class=\"dropzone-container__icon-upload icon-document-upload\"></i>\n            <p class=\"dropzone-container__label-dragdrop\">\n               Drag and Drop\n            </p>\n            <p class=\"dropzone-container__label-or\">\n                or\n            </p>\n          <fj-button\n            [icon]=\"'plus'\"\n            [iconMargin]=\"'0 5px 0px 0px'\"\n            [margin]=\"'0px'\"\n            [ghost]=\"true\"\n            [size]=\"'med'\"\n          >\n            Choose a local file\n          </fj-button>\n        </div>\n    </div>\n    <div class=\"drag-drop__updload-info\">\n        <fj-icon svgName=\"attachment\"\n                 fillColor=\"black\" class=\"drag-drop__updload-info-icon\"></fj-icon>\n        <span>Maximum file size is 20MB</span>\n    </div>\n</div>\n\n",
                    encapsulation: ViewEncapsulation.None,
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.drag-drop{display:flex;flex-direction:column;background-color:#f2f2f2;height:100%;width:100%;padding:1.75rem;box-sizing:border-box}.drag-drop__icon-right{float:right}.drag-drop__icon-right:hover{cursor:pointer}.drag-drop__file-missing{color:#db0000}.drag-drop__updload-info{display:inline-flex;align-items:center;margin-top:1.5625rem;font-size:.875rem;color:rgba(0,0,0,.45)}.drag-drop__updload-info-icon{margin-right:.4375rem}.dropzone-container{flex:1;justify-content:center;align-items:center;overflow:hidden;display:flex;position:relative;width:100%;cursor:pointer;background-color:rgba(0,0,0,.02);border:.0625rem dashed #f2f2f2;border-radius:.25rem;box-sizing:border-box;padding:.3125rem;text-align:center;line-height:.8125rem;transition:.3s}.dropzone-container span{display:block}.dropzone-container__icon-upload{font-size:2.5rem;color:#f2f2f2}.dropzone-container__placeholder{display:block;transition:transform .3s;transition:transform .3s,-webkit-transform .3s}.dropzone-container__label-dragdrop{margin-top:1.5625rem;color:rgba(0,0,0,.6)}.dropzone-container__label-or{color:rgba(0,0,0,.6)}.dropzone-container__btn-icon{margin-right:.5rem}.dropzone-container:hover{border:.0625rem dashed #1b51a0;background-color:#dce8f9}.dropzone-container:hover .dropzone-container__icon-upload{color:#1b51a0}.dropzone-container.dz-drag-hover{border:.0625rem solid #1b51a0;background-color:#dce8f9}.dropzone-container.dz-drag-hover .dropzone-container__placeholder{-webkit-transform:scale(.95);transform:scale(.95)}.dropzone-container.dz-drag-hover i{color:#1b51a0}"]
                }] }
    ];
    /** @nocollapse */
    FjDragAndDropComponent.ctorParameters = function () { return []; };
    FjDragAndDropComponent.propDecorators = {
        componentRef: [{ type: ViewChild, args: [DropzoneComponent,] }],
        directiveRef: [{ type: ViewChild, args: [DropzoneDirective,] }],
        uploadedDoc: [{ type: Output }],
        fileToUpload: [{ type: Output }],
        removeFile: [{ type: Input }],
        noFileSelected: [{ type: Input }]
    };
    return FjDragAndDropComponent;
}());
export { FjDragAndDropComponent };
