// freyja-ui/esm5/lib/content-switcher/content-switcher.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
var ContentSwitcherComponent = /** @class */ (function () {
    function ContentSwitcherComponent() {
        /**
         * content switcher options
         */
        this.options = [];
        /**
         * event triggered when an option is selected
         */
        this.optionSelected = new EventEmitter();
        /**
         * the id of the active tab
         */
        this.activeTab = 0;
    }
    ContentSwitcherComponent.prototype.ngOnInit = function () {
        var _this = this;
        if (this.selectedOption) {
            this.activeTab = this.options.findIndex(function (option) { return _this.selectedOption.toString() === option.value.toString(); });
        }
    };
    /**
     *
     * @param index - index of the selected element
     * @param value - value of the selected element
     */
    ContentSwitcherComponent.prototype.selectOption = function (index, value) {
        this.activeTab = index;
        var selectedElement = {
            index: index,
            value: value
        };
        this.optionSelected.emit(selectedElement);
    };
    ContentSwitcherComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-content-switcher',
                    template: "<div class=\"content-switcher content-switcher--radius\">\n  <button *ngFor=\"let option of options; let i = index\"\n          class=\"content-switcher__tab\"\n          [ngClass]=\"{'content-switcher__tab--is-active': activeTab === i}\"\n          (click)=\"selectOption(i, option.value)\">\n    {{option.label}}\n  </button>\n</div>\n",
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.content-switcher{display:flex;flex-wrap:wrap;font-family:open-sans,-apple-system,BlinkMacSystemFont,\"Segoe UI\",Roboto,\"Helvetica Neue\",Arial,sans-serif,\"Apple Color Emoji\",\"Segoe UI Emoji\",\"Segoe UI Symbol\",\"Noto Color Emoji\"}.content-switcher--rounded .content-switcher__tab:first-child{border-bottom-left-radius:2rem;border-top-left-radius:2rem}.content-switcher--rounded .content-switcher__tab:last-child{border-bottom-right-radius:2rem;border-top-right-radius:2rem}.content-switcher--radius .content-switcher__tab:first-child{border-right:0;border-bottom-left-radius:4px;border-top-left-radius:4px}.content-switcher--radius .content-switcher__tab:last-child{border-left:0;border-bottom-right-radius:4px;border-top-right-radius:4px}.content-switcher__tab{border:1px solid #1b51a0;color:#1b51a0;cursor:pointer;font-family:open-sans,-apple-system,BlinkMacSystemFont,\"Segoe UI\",Roboto,\"Helvetica Neue\",Arial,sans-serif,\"Apple Color Emoji\",\"Segoe UI Emoji\",\"Segoe UI Symbol\",\"Noto Color Emoji\";font-size:.875rem;height:40px;min-width:104px;outline:0}.content-switcher__tab:hover{background-color:#dce8f9}.content-switcher__tab--is-active{background-color:#1b51a0!important;color:#fff;pointer-events:none}.content-switcher--sm .content-switcher__tab{font-size:.8125rem;height:32px}.content-switcher--disabled .content-switcher__tab{border-color:rgba(0,0,0,.15);color:var(--color-gray);pointer-events:none}.content-switcher--disabled .content-switcher__tab--is-active{background-color:rgba(0,0,0,.15);border-color:transparent;color:#fff}"]
                }] }
    ];
    ContentSwitcherComponent.propDecorators = {
        options: [{ type: Input }],
        selectedOption: [{ type: Input }],
        optionSelected: [{ type: Output }]
    };
    return ContentSwitcherComponent;
}());
export { ContentSwitcherComponent };
