// freyja-ui/esm5/lib/light-box/light-box.component.js
import { Component, Input } from '@angular/core';
var LightBoxComponent = /** @class */ (function () {
    function LightBoxComponent() {
        /**
         * checks if the lightbox is open
         */
        this.isOpen = false;
        /**
         * current image slide index
         */
        this.currentSlideIndex = 1;
    }
    LightBoxComponent.prototype.ngOnInit = function () {
    };
    /**
     * opens the lightbox modal and set the current slide
     * slideNumber - the seleceted slide number
     */
    LightBoxComponent.prototype.openModal = function (slideNumber) {
        if (this.slides) {
            this.isOpen = true;
            this.currentSlide(slideNumber);
        }
    };
    /**
     * closes the lightbox modal
     */
    LightBoxComponent.prototype.closeModal = function () {
        this.isOpen = false;
    };
    // Next/previous controls
    LightBoxComponent.prototype.plusSlides = function (n) {
        this.showSlides(this.currentSlideIndex += n);
    };
    // Thumbnail image controls
    LightBoxComponent.prototype.currentSlide = function (n) {
        this.showSlides(this.currentSlideIndex = n);
    };
    /**
     * show the slides in the lightbox modal
     * @param n - slide number
     */
    LightBoxComponent.prototype.showSlides = function (n) {
        if (n > this.slides.length) {
            this.currentSlideIndex = 1;
        }
        else if (n < 1) {
            this.currentSlideIndex = this.slides.length;
        }
        this.slides.forEach(function (slide) {
            slide.hidden = true;
        });
        this.slides[this.currentSlideIndex - 1].hidden = false;
    };
    LightBoxComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-light-box',
                    template: "<!-- The Modal/Lightbox -->\n<div class=\"lightbox__modal\" [ngClass]=\"{'lightbox__modal__shown': isOpen}\" *ngIf=\"slides\">\n  <span class=\"lightbox__modal__close lightbox__modal__cursor\" (click)=\"closeModal()\">&times;</span>\n  <div class=\"lightbox__modal__content\">\n\n    <div class=\"lightbox__slides\" *ngFor=\"let slide of slides; let i = index\"  [ngClass]=\"{'lightbox__slides__hidden': slide.hidden}\">\n      <div class=\"lightbox__number\">{{i + 1}} / {{slides.length}}</div>\n      <div class=\"lightbox__image\" [ngStyle]=\"{'background-image': 'url(' + slide?.src + ')'}\">\n    </div>\n\n    <!-- Next/previous controls -->\n    <a class=\"lightbox__prev\" (click)=\"plusSlides(-1)\">&#10094;</a>\n    <a class=\"lightbox__next\" (click)=\"plusSlides(1)\">&#10095;</a>\n\n    <!-- Caption text -->\n    <div class=\"lightbox__caption-container\">\n      <p id=\"lightbox__caption\">{{slides[currentSlideIndex - 1]?.alt}}</p>\n    </div>\n\n    <!-- Thumbnail image controls -->\n    <div class=\"lightbox__row\">\n      <div class=\"lightbox__column\" *ngFor=\"let slide of slides; let j = index\">\n        <div class=\"lightbox__thumbnail\" [ngStyle]=\"{'background-image': 'url(' + slide?.src + ')'}\" (click)=\"currentSlide(j + 1)\" [ngClass]=\"{'lightbox__thumbnail__active': !slide.hidden}\">\n      </div>\n    </div>\n  </div>\n</div>\n",
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.lightbox__row{display:flex;background-color:#000}.lightbox__column{width:25%}.lightbox__modal{display:none;position:fixed;z-index:999;padding:3.75rem 0;box-sizing:border-box;left:0;top:0;width:100%;height:100%;overflow:auto;background-color:#000}.lightbox__modal__shown{display:block}.lightbox__modal__content{position:relative;background-color:#fefefe;margin:auto;padding:0;width:90%;max-width:75rem;height:100vh}.lightbox__modal__close{color:#fff;position:absolute;top:10px;right:25px;font-size:35px;font-weight:700}.lightbox__modal__close:focus,.lightbox__modal__close:hover{color:#999;text-decoration:none;cursor:pointer}.lightbox__slides{display:block;height:100%}.lightbox__slides__hidden{display:none}.lightbox__next,.lightbox__prev{cursor:pointer;position:absolute;top:50%;width:auto;padding:16px;margin-top:-50px;color:#fff;font-weight:700;font-size:20px;transition:.6s;border-radius:0 3px 3px 0;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-user-select:none}.lightbox__next:hover,.lightbox__prev:hover{background-color:rgba(0,0,0,.8)}.lightbox__next{right:0;border-radius:3px 0 0 3px}.lightbox__number{color:#f2f2f2;font-size:12px;padding:8px 12px;position:absolute;top:0}.lightbox__image{height:100%;width:100%;background-size:cover}.lightbox__caption-container{text-align:center;background-color:#000;padding:2px 16px;color:#fff}.lightbox__thumbnail{opacity:1;cursor:pointer;width:100%;height:9.375rem;background-size:cover}.lightbox__thumbnail:hover,.lightbox__thumbnail__active{opacity:.6}.lightbox__hover-shadow{transition:.3s}.lightbox__hover-shadow:hover{box-shadow:0 4px 8px 0 rgba(0,0,0,.2),0 6px 20px 0 rgba(0,0,0,.19)}"]
                }] }
    ];
    /** @nocollapse */
    LightBoxComponent.ctorParameters = function () { return []; };
    LightBoxComponent.propDecorators = {
        slides: [{ type: Input }]
    };
    return LightBoxComponent;
}());
export { LightBoxComponent };
