// freyja-ui/esm2015/lib/checkbox-grid/checkbox-grid.component.js
import { Component, Input, EventEmitter, Output } from '@angular/core';
import { FormGroup } from '@angular/forms';
export class CheckboxGridComponent {
    constructor() {
        this.inline = true;
        this.noPadding = false;
        this.noMargin = false;
        this.smallText = false;
        this.checkboxClicked = new EventEmitter();
    }
    ngOnChanges(changes) {
        this.makeCheckboxes();
    }
    /**
     * @description Create the internally used array of checkboxes
     */
    makeCheckboxes() {
        if (this.checkboxGroup) {
            this.checkboxes = this.checkboxGroup.map((item, index) => {
                return {
                    index: index,
                    id: `${item.value}_${index}`,
                    label: item.label,
                    value: item.value,
                    checked: item.checked || false,
                };
            });
        }
    }
    /**
     * @description Notify the parent that a value has changed
     * @param checked the new checked value of the checkbox
     * @param item the internal checkbox model
     */
    updateChecked(checked, item) {
        if (item.disabled) {
            return;
        }
        item.checked = checked;
        const toggledItem = this.checkboxGroup[item.index];
        if (toggledItem.hasOwnProperty('checked')) {
            toggledItem.checked = item.checked;
        }
        this.checkboxClicked.emit(toggledItem);
    }
}
CheckboxGridComponent.decorators = [
    { type: Component, args: [{
                selector: 'fj-checkbox-grid',
                template: "<h2 class=\"title\">Main Title</h2>\n<div class=\"grid-form__card-body card-body\" [ngClass]=\"{'no-padding': noPadding, 'no-margin': noMargin, 'small-text': smallText}\">\n  <div *ngIf=\"formGroup\" [formGroup]=\"formGroup\">\n    <div *ngFor=\"let checkbox of checkboxes;\" class=\"form-check\" [ngClass]=\"{'form-check--inline': inline}\">\n      <div class=\"custom-control custom-checkbox\">\n        <input\n          type=\"checkbox\"\n          class=\"custom-control-input\"\n          [value]=\"checkbox.value\"\n          [ngModel]=\"checkbox.checked\"\n          (ngModelChange)=\"updateChecked($event, checkbox)\"\n          [disabled]=\"checkbox.disabled\"\n          [attr.id]=\"checkbox.id\"\n          [attr.name]=\"checkbox.id\" />\n        <label class=\"custom-control-label\" [attr.for]=\"checkbox.id\">{{checkbox.label}}</label>\n      </div>\n    </div>\n  </div>\n  <div *ngIf=\"!formGroup\" [ngClass]=\"{'no-border': noBorder}\">\n    <div *ngFor=\"let checkbox of checkboxes; \" class=\"form-check\" [ngClass]=\"{'form-check--inline': inline}\">\n      <div class=\"custom-control custom-checkbox\">\n        <input\n          type=\"checkbox\"\n          class=\"custom-control-input\"\n          [value]=\"checkbox.value\"\n          [ngModel]=\"checkbox.checked\"\n          (ngModelChange)=\"updateChecked($event, checkbox)\"\n          [disabled]=\"checkbox.disabled\"\n          [attr.id]=\"checkbox.id\"\n          [attr.name]=\"checkbox.id\" />\n        <label class=\"custom-control-label\" [attr.for]=\"checkbox.id\">{{checkbox.label}}</label>\n      </div>\n    </div>\n  </div>\n</div>\n",
                styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}"]
            }] }
];
/** @nocollapse */
CheckboxGridComponent.ctorParameters = () => [];
CheckboxGridComponent.propDecorators = {
    title: [{ type: Input }],
    checkboxGroup: [{ type: Input }],
    inline: [{ type: Input }],
    noPadding: [{ type: Input }],
    noMargin: [{ type: Input }],
    smallText: [{ type: Input }],
    noBorder: [{ type: Input }],
    formGroup: [{ type: Input }],
    checkboxClicked: [{ type: Output }]
};
