// freyja-ui/esm2015/lib/native-dropdown/native-dropdown.component.js
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { BaseComponent } from '../shared/base.component';
import { FormGroup } from '@angular/forms';
let nextId = 0;
export class NativeDropdownComponent extends BaseComponent {
    constructor() {
        super();
        /**
         * options of the dropdown
         */
        this.options = [];
        /**
         * dropdown id
         */
        this.id = `fj-native-dd-${++nextId}`;
        /**
         * wether or not to show default value after selecting an option
         */
        this.showDefaultOnCLick = false;
        /**
         * event that is triggered when an option is selected
         */
        this.optionSelected = new EventEmitter();
    }
    /**
     * function triggered when an option is selected
     * @param evt - mouse click event
     */
    onOptionSelected(evt) {
        evt.stopPropagation();
        const dropdownElem = document.getElementById(this.id);
        this.optionSelected.emit(dropdownElem.value);
        if (this.showDefaultOnCLick) {
            // reselect default value
            dropdownElem.value = this.options[0].label;
        }
    }
    /**
     * @param evt - mouse click event
     */
    onDropdownClicked(evt) {
        evt.stopPropagation();
    }
    /**
     * @description Add css classes based on inputs like size, shape etc
     */
    generateClassList() {
        const baseCls = 'fj-native-dropdown--';
        const classes = {
            [baseCls + this.size]: this.size,
        };
        return classes;
    }
}
NativeDropdownComponent.decorators = [
    { type: Component, args: [{
                selector: 'fj-native-dropdown',
                template: "<div [formGroup]=\"formGroup\" *ngIf=\"formGroup\">\n  <label\n    [attr.for]=\"id\"\n    class=\"label--medium\"\n    *ngIf=\"label\"\n  >{{label}}</label>\n  <select class=\"fj-native-dropdown\"\n          [ngClass]=\"generateClassList()\"\n          id=\"{{id}}\"\n          (click)=\"onDropdownClicked($event)\"\n          (change)=\"onOptionSelected($event)\"\n          [ngStyle]=\"customStyles\"\n          formControlName=\"{{controlName}}\"\n  >\n    <option *ngFor=\"let option of options\" [ngValue]=\"option.value\">{{option.label}}</option>\n  </select>\n</div>\n<!-- Input without reactive forms-->\n<div *ngIf=\"!formGroup\">\n  <label\n    [attr.for]=\"id\"\n    class=\"label--medium\"\n    *ngIf=\"label\"\n  >{{label}}</label>\n  <select class=\"fj-native-dropdown\"\n          id=\"{{id}}\"\n          [ngClass]=\"generateClassList()\"\n          (click)=\"onDropdownClicked($event)\"\n          (change)=\"onOptionSelected($event)\"\n          [ngStyle]=\"customStyles\"\n  >\n    <option *ngFor=\"let option of options\" value=\"{{option.value}}\">{{option.label}}</option>\n  </select>\n</div>\n",
                styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-native-dropdown{-moz-appearance:none;-webkit-appearance:none;font-family:OpenSans,sans-serif;font-size:1rem;height:2.5rem;padding:0 .9375rem;background-color:#fff;color:#1b51a0;cursor:pointer}.fj-native-dropdown::-ms-expand{display:none}.fj-native-dropdown:hover{background-color:#f0f8ff}.fj-native-dropdown--large{font-size:1rem;height:3rem;line-height:1.25rem;padding:.875rem 1.25rem}.fj-native-dropdown--medium{font-size:.875rem;height:2.5rem;line-height:1.125rem;padding:.6875rem 1rem}.fj-native-dropdown--small{font-size:.75rem;height:2rem;line-height:1rem;padding:.5rem .75rem}.fj-native-dropdown--micro{font-size:.625rem;height:1.5rem;line-height:.75rem;padding:.375rem .5rem}label{cursor:pointer;font-size:.875rem}label.label--medium{font-size:.75rem;padding-left:.5em}label.label--disabled{color:rgba(0,0,0,.2)}"]
            }] }
];
/** @nocollapse */
NativeDropdownComponent.ctorParameters = () => [];
NativeDropdownComponent.propDecorators = {
    options: [{ type: Input }],
    id: [{ type: Input }],
    width: [{ type: Input }],
    showDefaultOnCLick: [{ type: Input }],
    optionSelected: [{ type: Output }],
    error: [{ type: Input }],
    controlName: [{ type: Input }],
    formGroup: [{ type: Input }]
};
