// freyja-ui/esm5/lib/radio-button/radio-button.component.js
import * as tslib_1 from "tslib";
// Angular Imports
import { Component, Input, EventEmitter, Output } from '@angular/core';
// Internal Imports
import { BaseComponent } from '../shared/base.component';
/**
 * Count for Next Id
 */
var nextId = 0;
/**
 * Freyja Radio Button Component according to specifics of Freyja of Freyja Design
 */
var RadioButtonComponent = /** @class */ (function (_super) {
    tslib_1.__extends(RadioButtonComponent, _super);
    function RadioButtonComponent() {
        var _this = _super.call(this) || this;
        // This is to make component controlled
        _this.checkedValue = false;
        _this.uniqueInputId = "fj-radio-button-" + ++nextId;
        /**
         * Provide unique ID or it will use auto generated from counter
         */
        _this.id = "fj-radio-button-" + ++nextId;
        /**
         * on change event, will be emitted once user clicks on radio button to change value
         */
        _this.change = new EventEmitter();
        return _this;
    }
    Object.defineProperty(RadioButtonComponent.prototype, "checked", {
        /**
         * Will determin initial checked state of radio button
         */
        get: function () {
            return this.checkedValue;
        },
        set: function (v) {
            if (v !== this.checkedValue) {
                this.checkedValue = this.coerceBooleanInput(v);
                this.change.emit(this.checkedValue);
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * Generate classes using BEM appraoch and bind to ngClass
     */
    RadioButtonComponent.prototype.generateClassList = function () {
        var _a;
        var base = 'fj-radio-btn--';
        var classes = (_a = {},
            _a[base + "label-before"] = this.labelPosition === 'before',
            _a[base + "disabled"] = this.disabled,
            _a);
        return classes;
    };
    /**
     * capture on click and toggle state of radio button
     * @param event emitted by html element on click
     */
    RadioButtonComponent.prototype.onRadioButtonClicked = function (event) {
        event.stopPropagation();
        this.toggle();
    };
    RadioButtonComponent.prototype.onChangeEvent = function (event) {
        event.stopPropagation();
    };
    /**
     * Toggle state of checked value and emit on change
     */
    RadioButtonComponent.prototype.toggle = function () {
        this.checked = !this.checked;
        this.change.emit(this.checkedValue);
    };
    RadioButtonComponent.decorators = [
        { type: Component, args: [{
                    selector: 'fj-radio-button',
                    template: "<label [attr.for]=\"id\"\n       class=\"fj-radio-btn\"\n       [ngClass]=\"generateClassList()\">\n  <input class=\"fj-radio-btn__input\"\n         type=\"radio\"\n         [id]=\"id\"\n         [disabled]=\"disabled\"\n         [tabIndex]=\"tabIndex\"\n         [attr.name]=\"name\"\n         [attr.aria-label]=\"ariaLabel\"\n         [attr.aria-labelledby]=\"ariaLabelledby\"\n         [attr.aria-describedby]=\"ariaDescribedby\"\n         [attr.value]=\"value\"\n         [checked]=\"checked\"\n         (change)=\"onChangeEvent($event)\"\n         (click)=\"onRadioButtonClicked($event)\">\n  <span class=\"fj-radio-btn__label\">\n    <ng-content></ng-content>\n  </span>\n</label>\n",
                    styles: ["@font-face{font-family:open-sans;font-weight:400;src:url(/assets/fonts/OpenSans-Regular.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Regular.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Regular.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Regular.svg#open-sans) format(\"svg\")}@font-face{font-family:open-sans;font-weight:600;src:url(/assets/fonts/OpenSans-Bold.eot?) format(\"eot\"),url(/assets/fonts/OpenSans-Bold.woff) format(\"woff\"),url(/assets/fonts/OpenSans-Bold.ttf) format(\"truetype\"),url(/assets/fonts/OpenSans-Bold.svg#open-sans) format(\"svg\")}.fj-radio-btn{display:inline-flex;cursor:pointer;white-space:nowrap;align-items:center;margin-right:.5625rem}.fj-radio-btn__input{display:inline-block;-webkit-appearance:none;-moz-appearance:none;appearance:none;cursor:pointer;height:1.125rem;outline:0;position:relative;width:1.125rem;margin-right:.5625rem}.fj-radio-btn__input::before{display:block;left:0;top:0;background-color:transparent;border:.0625rem solid #1b51a0;content:'';height:1rem;position:absolute;transition:.1s linear;width:1rem;border-radius:1rem}.fj-radio-btn__input::after{display:block;border-bottom:.125rem solid #fff;border-right:.125rem solid #fff;content:'';opacity:0;position:absolute;-webkit-transform:rotate(45deg);transform:rotate(45deg);border:.0625rem solid #1b51a0;border-radius:1em;background-color:#1b51a0;height:.5rem;left:.25rem;top:.25rem;width:.5rem}.fj-radio-btn__input:active::before,.fj-radio-btn__input:focus::before,.fj-radio-btn__input:hover::before{background-color:#dce8f9}.fj-radio-btn__input:checked::after{opacity:1}.fj-radio-btn__input:checked::before{background-color:transparent}.fj-radio-btn__input:checked:active::before,.fj-radio-btn__input:checked:focus::before,.fj-radio-btn__input:checked:hover::before{background-color:#dce8f9}.fj-radio-btn__input:disabled{opacity:.4;cursor:default}.fj-radio-btn__input:disabled::after,.fj-radio-btn__input:disabled:active::after,.fj-radio-btn__input:disabled:focus::after,.fj-radio-btn__input:disabled:hover::after{background-color:#000}.fj-radio-btn__input:disabled:active::before,.fj-radio-btn__input:disabled:focus::before,.fj-radio-btn__input:disabled:hover::before{background-color:transparent}.fj-radio-btn--label-before>input{order:1;margin-left:.5625rem}.fj-radio-btn--disabled{opacity:.5;cursor:auto}"]
                }] }
    ];
    /** @nocollapse */
    RadioButtonComponent.ctorParameters = function () { return []; };
    RadioButtonComponent.propDecorators = {
        id: [{ type: Input }],
        value: [{ type: Input }],
        change: [{ type: Output }],
        checked: [{ type: Input }]
    };
    return RadioButtonComponent;
}(BaseComponent));
export { RadioButtonComponent };
