// freyja-ui/esm5/lib/date-picker/date-picker.service.js
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
var DatePickerService = /** @class */ (function () {
    function DatePickerService() {
        this.getProgramDirectEntryCalendarStartDate = new Subject();
        this.getEditProgramCalendarStartDate = new Subject();
        this.getEditActivityCalendarStartDate = new Subject();
        this.getApprovalCalendarStartDate = new Subject();
        this.getCreateMilestoneCalendarStartDate = new Subject();
    }
    //  Method set disableUntil Start date value until yesterday
    DatePickerService.prototype.disableUntil = function (options) {
        var d = new Date();
        d.setDate(d.getDate() - 1);
        var copy = this.getCopyOfOptions(options);
        copy.disableUntil = {
            year: d.getFullYear(),
            month: d.getMonth() + 1,
            day: d.getDate()
        };
        var myOptions = copy;
        return myOptions;
    };
    // Returns copy of myOptions for the Date
    DatePickerService.prototype.getCopyOfOptions = function (options) {
        return JSON.parse(JSON.stringify(options));
    };
    DatePickerService.prototype.getEndDateCalendar = function (date, options) {
        var myEndDateOptions = this.getCopyOfOptions(options);
        myEndDateOptions.disableUntil = date;
        myEndDateOptions.todayBtnTxt = 'End Date';
        return myEndDateOptions;
    };
    DatePickerService.prototype.getStartDate = function (date, form) {
        try {
            var formatedDate = {
                year: date.split('.')[2],
                month: date.split('.')[1],
                day: date.split('.')[0]
            };
            if (form === 'edit_program_form') {
                this.editProgramSartDate = formatedDate;
                this.getEditProgramCalendarStartDate.next(this.editProgramSartDate);
            }
            if (form === 'edit_activity_form') {
                this.editActivitySartDate = formatedDate;
                this.getEditActivityCalendarStartDate.next(this.editActivitySartDate);
            }
            if (form === 'approval_form') {
                this.approvalStartDate = formatedDate;
                this.getApprovalCalendarStartDate.next(this.approvalStartDate);
            }
            if (form === 'create_milestone_form') {
                this.createMilestoneStartDate = formatedDate;
                this.getCreateMilestoneCalendarStartDate.next(this.createMilestoneStartDate);
            }
            if (form === 'program_direct_entry_form') {
                this.programDirectEntryStartDate = formatedDate;
                this.getProgramDirectEntryCalendarStartDate.next(this.programDirectEntryStartDate);
            }
            return true;
        }
        catch (error) {
            return false;
        }
    };
    DatePickerService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    DatePickerService.ctorParameters = function () { return []; };
    return DatePickerService;
}());
export { DatePickerService };
